### ISSUE TITLE
detect.py crashed with multiple video sources specified in list.treams

### ISSUE THREAD

**Issue by regall00** [ultralytics/yolov5#13607]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### Bug

I run detect.py with single/multiple video sources: perfectly works with single video, failed with 2+ video sources. Could you please help to debug? Thank you

-----------cmd:
 python3 ./detect.py --weight yolov5s.pt --source list.streams --nosave

-----------list.streams
/yolov5/videos/video1.mp4
/yolov5/videos/video1_cloned.mp4

----------- Log
0: 384x640 4 persons, 3 suitcases, 1 chair, 1 tv, 1: 384x640 3 persons, 1 dog, 1 chair, 2 tvs, 23.2ms
0: 384x640 5 persons, 3 suitcases, 1 chair, 1 tv, 1: 384x640 3 persons, 1 chair, 2 tvs, 23.3ms
0: 384x640 5 persons, 4 suitcases, 1 chair, 1 tv, 1: 384x640 3 persons, 1 chair, 1 tv, 23.4ms
0: 384x640 4 persons, 5 suitcases, 1 chair, 1 tv, 1: 384x640 3 persons, 1 chair, 1 tv, 23.1ms
WARNING ⚠️ Video stream unresponsive, please check your IP camera connection.
0: 384x640 5 persons, 4 suitcases, 1 chair, 1 tv, 1 cell phone, 1: 384x640 3 persons, 1 chair, 1 tv, 23.2ms
0: 384x640 1: 384x640 3 persons, 1 chair, 23.7ms
Speed: 0.8ms pre-process, 13.5ms inference, 3.4ms NMS per image at shape (1, 3, 640, 640)
FATAL: exception not rethrown
Fatal Python error: Aborted

Thread 0x0000ffff99e06ca0 (most recent call first):
  <no Python frame>

Extension modules: numpy.core._multiarray_umath, numpy.core._multiarray_tests, numpy.linalg._umath_linalg, numpy.fft._pocketfft_internal, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._mt19937, numpy.random.mtrand, numpy.random._philox, numpy.random._pcg64, numpy.random._sfc64, numpy.random._generator, torch._C, torch._C._fft, torch._C._linalg, torch._C._nested, torch._C._nn, torch._C._sparse, 
..., 
scipy.stats._mvn, scipy.stats._rcont.rcont, scipy.ndimage._nd_image, scipy.ndimage._rank_filter_1d, _ni_label, scipy.ndimage._ni_label, scipy.cluster._vq, scipy.cluster._hierarchy, scipy.cluster._optimal_leaf_ordering (total: 157)
Aborted (core dumped)

### Environment

-  YOLO: YOLOv5 🚀 2025-5-27 Python-3.10.12 torch-2.2.0 CUDA:0 (Orin, 62841MiB)
- OS: Ubuntu 22.04.4 LTS
- Python: Python 3.10.12

### Minimal Reproducible Example

```-----------work:
#### file: list.streams
/yolov5/videos/video1.mp4

#### cmd:
python3 ./detect.py --weight yolov5s.pt --source list.streams --nosave

```

```-----------failed:
#### file: list.streams
/yolov5/videos/video1.mp4
/yolov5/videos/video1_cloned.mp4

#### cmd:
python3 ./detect.py --weight yolov5s.pt --source list.streams --nosave

```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @regall00, thank you for reporting this issue with YOLOv5 🚀! This is an automated response to help you get the most relevant support as quickly as possible. An Ultralytics engineer will also review your report and assist you soon.

If this is a 🐛 Bug Report, please ensure you have provided a **minimum reproducible example** (MRE). From your description and logs, it looks like you've included the key files and commands—thank you! If you can further narrow down the problem or provide any additional error output, it would be helpful.

To help us debug and assist you more efficiently, please also verify:
- Your [Python>=3.8.0](https://www.python.org/) environment and [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) are up to date, including [PyTorch>=1.8](https://pytorch.org/get-started/locally/).
- You have tested in one of our recommended environments.

## Quickstart
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Verified Environments
YOLOv5 can be run in any of the following up-to-date verified environments (with all dependencies including CUDA/CUDNN, Python, and PyTorch preinstalled):
- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status
<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>
If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

For more information, please check our [Tutorials](https://docs.ultralytics.com/yolov5/), including guides for [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) and [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/).

Thank you for your detailed report and for helping to improve YOLOv5! 🛠️

**Comment by pderrenger** [comment]
The crash with multiple video sources appears to be related to memory or threading issues when processing concurrent video streams. Please ensure you're using the latest YOLOv5 version by running `git pull` or reinstalling with `pip install -U ultralytics`.

You can try these workarounds: reduce the image size with `--imgsz 320`, limit threads with `--max-det 100`, or process videos sequentially by running detect.py separately for each video source instead of using the list.streams file.

**Comment by regall00** [comment]
I suspect that the crash is not due to concurrent situation, but threads are not cleanly ended/handled: error happens when finishing read one video, the other video still has frames to read.

**Comment by pderrenger** [comment]
You're absolutely right about the thread cleanup issue - this appears to be a bug in how YOLOv5 handles thread termination when videos have different lengths or end times. This is a known limitation in the multi-source video processing code where threads don't properly synchronize when one stream ends before others.

As a temporary workaround, you could try adding `--max-det 1000` to give the model more buffer time, or consider using a separate script to process multiple videos sequentially. Would you mind opening a dedicated bug report for this thread cleanup issue so the development team can address it properly?
